interpreter/interpreter.go

```diff
@@ -56,9 +68,13 @@ func (itrp *Interpreter) Eval(node semantic.Node, scope Scope, importer Importer
 		}
 	}), node)
 
+	// reset side effect list
+	itrp.sideEffects = make([]SideEffect, 0)
 	if err := itrp.doRoot(node, scope, importer); err != nil {
 		return nil, err
 	}
+	// update global side effects
+	itrp.globalSideEffects = append(itrp.globalSideEffects, itrp.sideEffects...)
```

Comment:
Perhaps you wanted to update the global side effects before you reset the local one? As it stands right now, you'd be appending an empty array to the global side effects.

Reply:
Mmmm 🤔 , I don't think so. Tests pass successfully, indeed.

`doRoot` updates local side effects.